OpenEdXMobile/src/main/java/org/edx/mobile/view/custom/AuthenticatedWebView.java

```diff
@@ -155,10 +158,26 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
 
             @Override
             public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
-                if (request.getUrl().toString().contains("logout")) {
+                String overrideUrl = request.getUrl().toString();
+                if (overrideUrl.contains("logout")) {
                     forceLogoutUser();
                     return true;
                 }
+                if (overrideUrl.contains("show_screen_without_dismissing")) {
+                    if (request.getUrl().getQueryParameter("screen_name").contains("terms_of_service")) {
+                        getContext().startActivity(WebViewActivity.newIntent(getContext(),
+                                getContext().getResources().getString(R.string.terms_file_link),
+                                getContext().getResources().getString(R.string.terms_of_service_title)));
+                    } else if (request.getUrl().getQueryParameter("screen_name").contains("privacy_policy")) {
+                        getContext().startActivity(WebViewActivity.newIntent(getContext(),
+                                getContext().getResources().getString(R.string.privacy_file_link),
+                                getContext().getResources().getString(R.string.privacy_policy)));
+                    }
+                    return true;
+                } else if (overrideUrl.contains("dismiss")) {
+                    pageFinishCallback.onUrlOverriderClick(!overrideUrl.contains("screen_name=delete_account"));
+                    return true;
+                }
```

Comment:
create a method for these line of code, reuse it in `shouldOverrideUrlLoading` for android 6 and below.